docs/XBlocks.md

```diff
@@ -0,0 +1,217 @@
+    * Settings - no stage-specific settings
+    * Components:
+        * `Submissions Help Text` - static (developer-defined) help text instructing students to open Project Navigator
+            Submissions view to upload submissions. Help text contains a link that opens the view automatically.
+        * `Submission` - specifies one deliverable (aka submission) - see [Stage components](#stage-components) section 
+            for details.
+    * Completion criteria: completed when all the submissions are uploaded. Note that submissions are required on group
+        basis rather than on individual basis, so all the students in the group get this stage completed as soon as
+        any of them uploads last submission. It does not matter which student uploads the submission - all uploads
+        count towards entire group progress.
+* `Team Evaluation` - allows students to provide anonymous feedback to their teammates.
+    * Settings - no stage-specific settings
+    * Components:
+        * `Teammate selector` - displays a list of teammates. **Required, one at most**
+        * `Grade Rubric Help Text` - static (developer-defined) help text instructing students to open Project Navigator
+            Resources view to access group project documetns on grading. Help text contains a link that opens the view 
+            automatically.
+        * `Review Question` - represents review question - see [Stage components](#stage-components) section for details
+    * Completion criteria: completed when student provided answers to all required questions for all teammates.
+* `Peer Grading` - allows students and/or TAs to provide anonymous feedback and grades to other groups.
+    * Settings - no stage-specific settings
+    * Components:
+        * `Group selector` - displays a list of groups student should grade. **Required, one at most**
+        * `Grade Rubric Help Text` - static (developer-defined) help text instructing students to open Project Navigator
+            Resources view to access group project documetns on grading. Help text contains a link that opens the view 
+            automatically.
+        * `Review Question` - represents review question - see [Stage components](#stage-components) section for details
+    * Completion criteria: completed when student provided answers to all required questions for all groups assigned 
+        for review.
+* `Evaluation Display` - displays teammate feedback for students
+    * Settings - no stage-specific settings
+    * Components:
+        * `Team Evaluation Display` - block that displays teammates' answers to one of the review questions. 
+            See [Stage components](#stage-components) section for details.
+    * Completion criteria: completed when user first visits it when the stage is open and teammates have competed their
+        reviews - provided answers to all the required questions chosen in `Team Evaluation Display` blocks.
```

Comment:
@e-kolpakov "competed" ⟶ "completed"

When first reading this it wasn't clear to me what the last part of the sentence meant. Maybe replace "reviews - provided" with "reviews, i.e., when teammates have provided".